src/System.ComponentModel.TypeConverter/src/System/ComponentModel/MemberDescriptor.cs

```diff
@@ -418,16 +459,27 @@ public abstract class MemberDescriptor {
         ///       Finds the given method through reflection.
         ///    </para>
         /// </devdoc>
-        protected static MethodInfo FindMethod(Type componentClass, string name, Type[] args, Type returnType, bool publicOnly) {
+        protected static MethodInfo FindMethod(Type componentClass, string name, Type[] args, Type returnType, bool publicOnly)
+        {
             MethodInfo result = null;
 
-            if (publicOnly) {
-                result = componentClass.GetMethod(name, args);
-            }
-            else {
-                result = componentClass.GetMethod(name, BindingFlags.Instance | BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic, null, args, null);
+            if (publicOnly)
+            {
+                result = componentClass.GetTypeInfo().GetMethod(name, args);
             }
-            if (result != null && !result.ReturnType.IsEquivalentTo(returnType)) {
+            else
+            {
+                // The original impementation requires the method https://msdn.microsoft.com/en-us/library/5fed8f59(v=vs.110).aspx which is not 
+                // available on .NET Core. The replacement will use the default BindingFlags, which may miss some methods that had been found
+                // on .NET Framework.
+#if FEATURE_GETMETHOD_WITHTYPES
+                result = componentClass.GetTypeInfo().GetMethod(name, BindingFlags.Instance | BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic, null, args, null);
+#else
+                result = componentClass.GetTypeInfo().GetMethod(name, args);
+#endif
+            }
+            if (result != null && !result.ReturnType.GetTypeInfo().IsEquivalentTo(returnType))
```

Comment:
There's an overload that takes just a Type. We have a work item to go through and clean up reflection usage more, but don't want to make too many changes with an already too large PR :)